misc.go

```diff
@@ -13,12 +13,26 @@ func Bool() bool {
 	return randIntRange(0, 1) == 1
 }
 
-// UUID (version 4) will generate a random unique identifier based upon random nunbers
+// UUID (version 4) will generate a random unique identifier based upon random numbers
 // Format: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
 func UUID() string {
+	return uuid(nil)
+}
+
+// UUID (version 4) will generate a random unique identifier based upon random numbers
+// Format: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+func (f Faker) UUID() string {
+	return uuid(f.Rand)
+}
+
+func uuid(r *rand.Rand) string {
 	version := byte(4)
 	uuid := make([]byte, 16)
-	rand.Read(uuid)
+	if r != nil {
+		r.Read(uuid)
+	} else {
```

Comment:
Would `r != nil` ever be false? I mean from line 25 you pass `f.Rand` which is initialised in the 'constructor', is that not enough?

Reply:
For this PR's potential way of doing things, https://github.com/brianvoe/gofakeit/pull/130/files#diff-d9ee104a2d1ced65b097da64fa502a9b6487f6c65ed0fca5d5c88a13da9c5abeR19